homeassistant/components/fibaro/quality_scale.yaml

```diff
@@ -0,0 +1,77 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  appropriate-polling:
+    status: exempt
+    comment: |
+      This integration uses a push API. No polling required.
```

Comment:
Is it fully push? Because I am wondering why we still use the `update` function (which only is run on boot since `update_on_add` is set to True)

Reply:
It is fully push, the push channel opened and managed by the FibaroController calls update on the entities through the callbacks which are registered by the entities during entity setup.
Based on your comment I think that this is wrong? What is the proposed way by HA?

Reply:
I think it depends on when you want that code to be run. If that should only happen at boot, we could either do update or have a function in async_added_to_hass. But I am mostly wondering what that code does

Reply:
`update()` is called by HA, this is one of the core concept of entities I guess.
1. In `async_added_to_hass` the `FibaroEntity` registers the callback which then triggers `schedule_update_ha_state(force = True)`
2. When a state update for this entity is sent by the fibaro hub, the `FibaroController` updates the `FibaroDevice` object with the updated state
3. The `FibaroController` executes then the callback on the `FibaroEntity` (which in turn calls then `schedule_update_ha_state(force = True)`)
4. HA calls then `update()` which we use to update the entity state based on the `FibaroDevice` object which was updated by the `FibaroController` before

So in the end the `update()` method runs on boot and then on each state update.

If we like to rewrite, we should probably copy the needed state attributes from `FibaroDevice` to the `FibaroEntity` and register an event which not only is just a trigger, but also contains the updated state attributes so that the data is not exchanged through the double references `FibaroDevice` object.

What do you think, is the old mechanism ok or should we change that?